docs/design/editorial_design.md

```diff
@@ -0,0 +1,354 @@
+
+> Again, no crazy about the argument names
+
+---
+
+
+## Roll
+![Roll](../_static/edit/09_roll.png)
+- Any trim-like action results in adjacent items source_range being adjusted to fit
+- No new items are ever created
+- Clamped to available media (if available)
+
+#### API
+````py
+"""
+Args:
+    item: The item to initiate the edit on
+    delta_in: see trim()
+    delta_out: see trim()
+"""
+otio.algorithms.roll(
+    item: Item,
+    delta_in: RationalTime = None,   #< Duration of change
+    delta_out: RationalTime = None,  #< Duration of change
+)
+````
+- In the example image, `A` is the item and `delta_out` is a negative RationalTime. The edit command augments `B` without changing it's `source_range().end_time_exclusive()`
+- If `delta_in` is supplied, the item to the left of `A` would be modified (if any) - otherwise we may need to add a `Gap` / fill template.
+
+---
+
+## 3/4 Point Edit
+![3_4 Point Edit](../_static/edit/08_pointedit.png)
+- The most complex - this "fills" a gap based on a source in/out point _or_ track in/out point
+- Often used to patch in items as edit is built
+- Note: This can be accomplished by a conjunction of commands above
+
+### API
+```py
+"""
+Args:
+    item: The item to place onto the track
+    track: Track that will now own this item
+    replace: Item (or RT?) that we will effectively replace
+    reference_point: For 4 point editing, the reference point dictates what
+        transform to use when running the fill.
+
+        Options:
+            - Source   : Don't modify the source, overwrite if required
+            - Sequence : Don't modify the sequence, trim item if required
+            - Fit      : Apply Time Effect
+"""
+otio.algorithms.fill(
+    item: Item,
+    track: Track,
+    replace: Item = None, # Alternatively, we could have a parent-coord
+                          # RationalTime to find the item for us
+    reference_point: otio.algorithms.ReferencePoint.Source
+)
+```
+- This may still require some additional tinkering.
+
+
+# Expanded Implementation
+
+## Proposal
+For editing commands, we should strive to do all validation and assert that everything "works" before committing anything on the timeline. A simple atomic command structure will provide that level of sophistication.
+
+```cpp
+class EditEvent {
+public:
+    EditEventKind kind; // e.g. Insert, Append, Remove, Modify
+    Retainer<Composition> parent;
+    Retainer<Item> composable;
+
+    // ... Additional fields to execute the above as required
+
+    bool run(ErrorStatus* error_status);
+    bool revert();
+};
+```
+An event is atomic in nature and does "one thing" to an `Item`. Each edit maneuver (e.g. `otio.algorithms.overwrite(...)`) would generate a vector of these events that can be played forward and, possibly, backward. The result of them collectively is the commands result.
```

Comment:
Lets factor this out into a second PR.  It is an orthogonal concept and also relevant to other non-edit api operations in OTIO, so we should design that as its own piece.

Reply:
I totally see it being used more widely. I was hoping to build the common commands on top of this system. A command like `overwrite` will potentially do quite a few events - something like "sub-edits"
```
e.g.
- set A.duration = F
- set B.duration = B
- insert C into T' at index 3
```
If the general feeling is they shouldn't use it, then I'll write them as direct procedures.